src/schema/rules/datatypes/nirs.yaml

```diff
@@ -0,0 +1,63 @@
+---
+- suffixes:
+  - nirs
+  extensions:
+  - .snirf
+  - .json
+  entities:
+    subject: required
+    session: optional
+    task: required
+    run: optional
+
+- suffixes:
+  - channels
+  extensions:
+  - .tsv
+  entities:
+    subject: required
+    session: optional
+    task: required
+    run: optional
+
+- suffixes:
+  - coordsystem
+  extensions:
+  - .json
+  entities:
+    subject: required
+    session: optional
+
+- suffixes:
+  - optodes
+  extensions:
+  - .tsv
+  entities:
+    subject: required
+    session: optional
```

Comment:
you may want to add `acq` for cases where positions are measured multiple times, see from the [eeg spec](https://bids-specification.readthedocs.io/en/bep030/04-modality-specific-files/03-electroencephalography.html):

> The [acq-<label>](https://bids-specification.readthedocs.io/en/bep030/99-appendices/09-entities.html#acq) key/value pair can be used to indicate acquisition of the same data. For example, this could be the recording of electrode positions with a different electrode position recording device, or repeated digitization before and after the recording.

Reply:
Thanks for highlighting this. Based on this use case I can absolutely understand why we would add an acq. I will add it now, thanks.

Reply:
Should I be adding space too?

Reply:
space is for derived data that has been coregistered to another coordinate system. We don't use it for raw data, so I don't see the need for NIRS at this moment.

Reply:
Thanks @robertoostenveld